src/Lucene.Net/Index/SegmentReader.cs

```diff
@@ -208,20 +214,21 @@ internal TermInfosReader GetTermsReader()
 			}
 			
 			internal bool TermsIndexIsLoaded()
-			{
-				lock (this)
-				{
-					return tis != null;
-				}
+            {
+                // If this is null, we call LoadTermsIndex, which uses a lock.

```

Comment:
Please note that `LoadTermsIndex` has a comment "This method is not fully thread safe, and relies on the synchronization in IndexWriter". I wonder if we can relay on  `LoadTermsIndex` and the lock we have there, especially after the change you made there.

Reply:
The lock is there because this method isn't thread safe.

Reply:
The `lock` was added there along with the comment.
Maybe I'm missing something but this the lock makes this method thread safe.